lib/galaxy/workflow/modules.py

```diff
@@ -805,7 +805,8 @@ def get_runtime_inputs(self, connections=None):
             if step.type == "tool":
                 assert isinstance(step.module, ToolModule)
                 tool = step.module.tool
-                assert tool
+                if tool is None:
```

Comment:
Why does the preview route fill in runtime inputs ? This seems unnecessary, and not appropriate for e.g. creating an invocation report

Reply:
That's an interesting point. It happens here during state injection: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/managers/workflows.py#L1065. I would assume the reason could be to rigorosly validate the final state, but that might not be necessary for the preview.